### DISCUSSION TITLE
image crop size error

### DISCUSSION THREAD

**Discussion by qwang0225** [Project-MONAI/MONAI#7288] [Q&A]
Hi, I have preprocessed the dataset to contain a dictionary of "image", "label" where the size is standardized to be 1 * 512*512*64, I tried to use `RandCropByPosNegLabeld` to create `ROI` of 128*128*64, but keep getting an error. Does someone know what's causing the error? Thanks. 
`
```
train_transforms = Compose(
    [
        ScaleIntensityRanged(
            keys=["image"],
            a_min=-35,
            a_max=95,
            b_min=0.0,
            b_max=1.0,
            clip=True,
        ),
            ScaleIntensityRanged(
            keys=["label"],
            a_min=0.0, a_max=255.0,
            b_min=0.0, b_max=1.0,  # New intensity range
            clip=True,
        ),
        # CropForegroundd(keys=["image", "label"], source_key="image"),
        Orientationd(keys=["image", "label"], axcodes="RAS"),
        #Spacingd(keys=["image", "label"], pixdim=(1.5, 1.5, 1.0), mode=("bilinear", "nearest")),
        RandCropByPosNegLabeld(
            keys=["image", "label"],
            label_key="label",
            spatial_size=(128, 128, 64),
            pos=1,
            neg=1,
            num_samples=4,
            image_key="image",
            image_threshold=0,
        ),
       
```

`ValueError: The size of the proposed random crop ROI is larger than the image size, got ROI size (128, 128, 64) and label image size (512, 512, 2) respectively.`

**Comment by KumoLiu** [comment]
Hi, @qwang0225, thanks for your interest here.
From the error message, seems the specified ROI size is larger than the image size.
Could you please check the size before `RandCropByPosNegLabeld`?
Thanks!

**Reply by qwang0225** [reply]
Yes, I did. All all input sizes are (512, 512, slices), slice is a number between 32 to 40s, I don't have an image of (512, 512, 2). 
I tried different spacing and got errors.  

What's I'm trying to do is to increase the sample size. Since my model performs poor with sample size (64, 64, 64) and sample number=4 or 8. It seems (64,64,64) is the largest I can do. But an image of (512, 512, 32) should be able to be cropped based on a larger sample size. 
```
 Spacingd(keys=["image", "label"], pixdim=(1.5, 1.5, 3.0), mode=("bilinear", "near
 RandCropByPosNegLabeld(
     keys=["image", "label"],
     label_key="label",
     spatial_size=(96, 96, 96),
     pos=7,
     neg=1,
     num_samples=8,
     image_key="image",
     image_threshold=0,
 ),
```
`ValueError: The size of the proposed random crop ROI is larger than the image size, got ROI size (96, 96, 96) and label image size (149, 149, 54) respectively.`

**Reply by KumoLiu** [reply]
Yes, you could set `allow_smaller=True` in `RandCropByPosNegLabeld`.
https://github.com/Project-MONAI/MONAI/blob/88f8dd275dbe8ca856b45ca34d6d01f4ea4c0128/monai/transforms/croppad/dictionary.py#L922
